IPQ 04: Bipartition Graph

char-adadev opened the pull request:
The purpose of this PR is to introduce the Bipartition Graph IPQ.

I did my best to guarantee mutual dislike for each test case, please let me know if you notice any mismatch 🙏🏾 

For review:
* Please run learn preview
* Verify the example implementation passes tests
* Any suggestions for wording, grammar, etc are welcome

kyra-at-ada commented on 04/implement-bipartition-graph.checkpoint.md:
```diff
@@ -0,0 +1,348 @@
+Output: True
+
+Explanation:
+Fido can be placed in Group 1.
+Nala can be placed in Group 1.
+Cooper can be placed in Group 2.
+Spot can be placed in Group 2.
+Bruno can be placed in Group 1.
+
+None of the pups who would fight with each other are placed in the same group.
```
Same here; just ensure any changes from `bipartition-graph.md` are copied over to this document.

kyra-at-ada reviewed (approved):
Nice work! I left a few suggestions about wording and maybe providing some hints, but even without the changes I think this is good to go!

I drew out the graph for each test and confirmed they were undirected. Preview looks good and sample solution works.

One last note - should we provide a hint that lets them know this can be solved using a modified version of either BFS or DFS? Or maybe prompts them to think about that question for themselves?

char-adadev commented:
Thank you for your suggestions! I updated the description and changed some of the dog names. I also added some hints and a resource for learning more about bipartite graphs. I'll go ahead and merge, if you notice anything out of place please lemme know and I can fix-forward 🙏🏾
